pkg/controller/core/workload_controller.go

```diff
@@ -193,41 +191,51 @@ func (r *WorkloadReconciler) getClusterQueueAdmissionChecks(ctx context.Context,
 	return queue.Spec.AdmissionChecks, nil
 }
 
+func (r *WorkloadReconciler) reconcileSyncAdmissionChecks(ctx context.Context, wl *kueue.Workload, cqName string) (bool, error) {
+	// because we need to react to API cluster queue events, the list of checks from a cache can lead to race conditions
+	if queueAdmissionChecks, err := r.getClusterQueueAdmissionChecks(ctx, cqName); err == nil {
```

Comment:
unroll the function `getClusterQueueAdmissionChecks` here. It's too short to be worth it's own function.
Also, if there is an error getting the checks, we should return so that we can retry.

Reply:
done